### PR TITLE
InviteToken Source Addition

### PR DESCRIPTION
**Pull request by petedejoy**
We now have the enum `InviteSource` being stored in the `InviteToken` table. This allows us to determine whether an invite is system-level or workspace-level. It also fixes the issue we were experiencing with the Unified Tracking project.

### CODE REVIEW

File: database/datamodel.graphql [hunk 1]
```diff
@@ -76,6 +81,7 @@ type InviteToken {
   id: ID! @id
   email: String!
   token: String! @unique
+  source: InviteSource
```
Comment (schnie) [inline database/datamodel.graphql]: @petedejoy could make this `!`

File: src/routes/v1/oauth/callback/post.js [hunk 2]
```diff
@@ -65,6 +65,7 @@ export default async function(req, res, next) {
   );
 
   // Set the userId, either the existing, or the newly created one.
+  console.log(state);
```
Comment (schnie) [inline src/routes/v1/oauth/callback/post.js]: @petedejoy

File: src/lib/users/index.js [hunk 3]
```diff
@@ -77,7 +79,10 @@ export async function createUser(opts) {
   if (haveInvite) {
     // Return the invite ID from the database to use in the
     // user record if the user has an invite
-    mutation.id = await prisma.inviteToken({ token: inviteTokens }).id();
+    const inviteToken =
+      find(inviteTokens, { source: INVITE_SOURCE_SYSTEM }) ||
+      head(inviteTokens);
```
Comment (schnie) [inline src/lib/users/index.js]:
```suggestion
      head(inviteTokens);

```

File: src/lib/users/index.js [hunk 4]
```diff
@@ -77,7 +79,11 @@ export async function createUser(opts) {
...
+
+    mutation.id = inviteToken.id;
```
Comment (schnie) [inline src/lib/users/index.js]: @petedejoy maybe a comment on each of these lines describing whats going on. 

This logic is getting more complicated, so might be worth it to say it in english so the next person knows why.

File: src/routes/v1/oauth/callback/post.js (hunk 2 above)
Comment (petedejoy) [reply src/routes/v1/oauth/callback/post.js]: NOOOO I WAS DOING SO WELL

File: src/lib/users/index.js (hunk 4 above)
Comment (petedejoy) [reply src/lib/users/index.js]: Good call- fixing that up now

Comment (petedejoy) [reply src/lib/users/index.js]: Added some- let me know if you think they're overly verbose.

**Comment by schnie** [comment]
@petedejoy looking great. Only thing left might be to add a test or two to make sure we're handling that logic correctly in the post route.

Maybe a way to check that we correctly select a system token if we do pass it one and another to test that it falls back correctly or something, probably mocking the function that returns the array of tokens.

**Comment by petedejoy** [comment]
@schnie- makes sense! do you mean in the `createUsers` function? We build the array and set the token there, haven't changed the post route at all I don't think.

**Comment by schnie** [comment]
Yes, my bad. I'm thinking some more tests on the `createUsers` function. I was thinking you could mock up `generateRoleBindings` to add some different combinations of rolebindings and test that this logic behaves as you expect.

**Comment by petedejoy** [comment]
@schnie we should be good to go here?

**Pull request merged**